k4FWCore/components/Writer.cpp

```diff
@@ -220,15 +225,16 @@ class Writer final : public Gaudi::Functional::Consumer<void(const EventContext&
     // deleted by the store (and later deleted by the Frame, triggering a double
     // delete)
     for (const auto& coll : frameCollections) {
+      debug() << "Taking ownership of collection " << coll << " from the IOSvc as it belongs to a Frame" << endmsg;
       DataObject* storeCollection;
       if (m_dataSvc->retrieveObject("/Event/" + coll, storeCollection).isFailure()) {
-        error() << "Failed to retrieve collection " << coll << endmsg;
-        return;
+        debug() << "Failed to retrieve collection " << coll << " from data service" << endmsg;
+        continue;
       }
       // We take ownership back from the store
       if (m_dataSvc->unregisterObject(storeCollection).isFailure()) {
-        error() << "Failed to unregister collection " << coll << endmsg;
-        return;
+        debug() << "Failed to unregister collection " << coll << " from data service" << endmsg;
+        continue;
```

Comment:
```suggestion
        error() << "Failed to unregister collection " << coll << " from data service" << endmsg;
        return;
```

This should be independent of the previous one (it's impossible to be here if the collection is not found) so I think this should remain an error, something is probably very wrong if unregistering something that is in the store fails.